[CI] MixedClusterClientYamlTestSuiteIT 'Create a snapshot and then restore it' failures

**Build scan**:
https://gradle-enterprise.elastic.co/s/m62agbvicmhiy
https://gradle-enterprise.elastic.co/s/74gmzvgqiplla
https://gradle-enterprise.elastic.co/s/psp5un562yyzo

**Repro line**:
./gradlew ':qa:mixed-cluster:v7.0.0#mixedClusterTest' --tests "org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" \
  -Dtests.method="test {p0=snapshot.restore/10_basic/Create a snapshot and then restore it}" \
  -Dtests.seed=C48535D3AF9BDB85 \
  -Dtests.security.manager=true \
  -Dtests.locale=fr \
  -Dtests.timezone=America/Denver \
  -Druntime.java=8

**Reproduces locally?**:
No

**Applicable branches**:
7.x, 7.10

**Failure history**:
Quite a few since 2020/10/02

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(build.branch,build.build-scans),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'%22Create%20a%20snapshot%20and%20then%20restore%20it%22'),sort:!(process.time-start,desc))

**Failure excerpt**:
```
java.lang.AssertionError: Failure at [snapshot.restore/10_basic:54]: field [test_index.shards.0.type] is null
	at __randomizedtesting.SeedInfo.seed([C48535D3AF9BDB85:4CD10A090167B67D]:0)
	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:414)
	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:391)
```

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/snapshot.restore/10_basic.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/snapshot.restore/10_basic.yml
@@ -24,10 +24,6 @@ setup:
 ---
 "Create a snapshot and then restore it":
 
-  - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/64152"
-
   - do:
       snapshot.create:
         repository: test_repo_restore_1

--- a/server/src/main/java/org/elasticsearch/snapshots/RestoreService.java
+++ b/server/src/main/java/org/elasticsearch/snapshots/RestoreService.java
@@ -386,10 +386,17 @@ public class RestoreService implements ClusterStateApplier {
                                             aliases.add(alias.value);
                                         }
                                     }
-                                    indexMdBuilder.settings(Settings.builder()
-                                        .put(snapshotIndexMetadata.getSettings())
-                                        .put(IndexMetadata.SETTING_INDEX_UUID, currentIndexMetadata.getIndexUUID())
-                                        .put(IndexMetadata.SETTING_HISTORY_UUID, UUIDs.randomBase64UUID()));
+                                    final Settings.Builder indexSettingsBuilder = Settings.builder()
+                                            .put(snapshotIndexMetadata.getSettings())
+                                            .put(IndexMetadata.SETTING_INDEX_UUID, currentIndexMetadata.getIndexUUID());
+                                    // Only add a restore uuid if either all nodes in the cluster support it (version >= 7.9) or if the
+                                    // index itself was created after 7.9 and thus won't be restored to a node that doesn't support the
+                                    // setting anyway
+                                    if (snapshotIndexMetadata.getCreationVersion().onOrAfter(Version.V_7_9_0) ||
+                                            currentState.nodes().getMinNodeVersion().onOrAfter(Version.V_7_9_0)) {
+                                        indexSettingsBuilder.put(SETTING_HISTORY_UUID, UUIDs.randomBase64UUID());
+                                    }
+                                    indexMdBuilder.settings(indexSettingsBuilder);
                                     IndexMetadata updatedIndexMetadata = indexMdBuilder.index(renamedIndexName).build();
                                     rtBuilder.addAsRestore(updatedIndexMetadata, recoverySource);
                                     blocks.updateBlocks(updatedIndexMetadata);

